src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -148,11 +154,36 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
                     postOffice={this.postOffice}
                 />
                 {progressBar}
+                {this.renderBreadcrumb()}
                 {this.state.totalRowCount > 0 && this.state.styleReady && this.renderGrid()}
             </div>
         );
     };
 
+    private renderBreadcrumb() {
+        if (this.state.fileName) {
+            let breadcrumbText = this.state.variableName;
+            if (this.state.originalVariableShape) {
+                breadcrumbText += ' (' + this.state.originalVariableShape?.join(', ') + ')';
+            }
+            return (
+                <div className="breadcrumb-container control-container">
+                    <div className="breadcrumb">
+                        <div className="icon-python breadcrumb-file-icon" />
+                        <span>{this.state.fileName}</span>
```

Comment:
Looking at the screen sample, i can't see how or where this is displayed.
Not sure I'd tie the file name to a variable either.

Reply:
It's at the very top of the screenshot. Basically Jill wants us to display breadcrumbs in the data viewer webview showing which file the variable was declared in, the same way that text editors in VS Code have breadcrumbs.